unthink-stack/src/server/nunjucks-renderer.ts

```diff
@@ -8,10 +10,40 @@ const nunjucksContext = {
   'IS_PRODUCTION': config.isProduction
 };
 
-export function renderTemplateWithContextAdded(template: string, value: unknown): string {
-  const valueCast = value as object | undefined;
-  const fullValue = { ...valueCast, ...nunjucksContext};
-  const result = nunjucksEnvironment.render(template as string, fullValue);
+function render(template: string, value: object | undefined): string {
+  const fullValue = { ...value, ...nunjucksContext};
+  const rendered = nunjucksEnvironment.render(template, fullValue);
 
-  return result;
+  return rendered;
+}
+
+export function renderTemplateWithContextAdded(result: Result, _ctx: RouteContext): string {
+  if (result instanceof ViewResult) {
+    return render(result.template as string, result.value as object);
+  }
+
+  if (result instanceof MiddlewareResult) {
+    // Only supporting error codes.
+    //
+    // Successful status codes should not make it to this function
+    let template: string;
+    switch (result.status) {
+    case 400:
+    case 500:
+      template = 'error.njk';
+      break;
+    case 401:
+      template = 'unauthorized.njk';
+      break;
+    case 404:
+      template = 'not-found.njk';
+      break;
+    default:
+      throw new Error(`A default template for status code ${result.status} is not supported in the render function`);
+    }
+
+    return render(template, result.value as object);
+  }
```

Comment:
@lxndrdagreat this block hasn't been tested yet but should handle the following results from middleware (view or agnostic):

- error
- unauthorized
- notFound

I actually don't think `500` will call the render function in any of the route handlers - occurred to me as I'm writing this. Let me verify that and we may want to talk about that. 

Just thinking out loud but now that the render function is aware of status code we could render a generic "Error" page on 500s which are unexpected errors for both View and Middleware results.